W-17700088-outbound-policies-fa (#757)
mulesoft/docs-gateway: gateway/1.9/modules/ROOT/pages/policies-outbound-upstream-idle-timeout.adoc

= Upstream Idle Timeout Policy
ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]
:imagesdir: ../assets/images
:keywords: basic authentication, api gateway, flex gateway, gateway, policy

[width="100%", cols="5,15"]
|===
>s| Policy Name | Stream Idle Timeout

> >s|Summary | Sets a maximum duration a stream between Flex Gateway and a upstream service can remain idle

Reviewer: ```suggestion
>s|Summary | Sets the maximum idle duration for a stream between the Flex Gateway and an upstream service
```

>s|Category | Quality of Service
>s|First Flex Gateway version available | v1.9.0
^s| Returned Status Codes | 504 - Upstream request timeout
|===

== Summary

> The Upstream Idle Timeout policy enables you to set a maximum duration a stream between Flex Gateway and the upstream service can remain idle between requests. This policy prevents idle connections from consuming resources while allowing persistent connections to remain open. The timeout time starts at the last request. 

Reviewer: ```suggestion
The Upstream Idle Timeout policy lets you set a maximum idle duration between requests for a stream between Flex Gateway and the upstream service. By applying the policy, you prevent idle connections from consuming resources while simultaneously allowing persistent connections to remain open. The timeout starts at the last request.
```

> By default, all Flex Gateway API instances have an upstream idle timeout of 60 seconds. Apply the policy overrides this default.

Reviewer: ```suggestion
By default, all Flex Gateway API instances have an upstream idle timeout of 60 seconds. Apply the policy to override the default timeout.
```

== Configuring Policy Parameters

=== Flex Gateway Local Mode

In Flex Gateway Local Mode, you apply the policy to your API via declarative configuration files. Refer to the following policy definition and table of parameters:

----
- policyRef:

>     name: upstream-idle-timeout

Reviewer 2: correct name: idle-timeout-flex

  config:
    timeout: number // REQUIRED
----

[%header%autowidth.spread,cols="a,a,a,a"]
|===
|Parameter | Required or Optional | Default Value | Description

| `timeout`
| Required
| 60
| Time in seconds before Flex Gateway closes the connection.

|===

==== Resource Configuration Example

----
- policyRef:

>     name: upstream-timeout-flex

Reviewer 2: name: idle-timeout-flex
